third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/closures/bug_16198.move

```diff
@@ -0,0 +1,31 @@
+//# publish
+module 0xc0ffee::m {
+    struct Func(|| | |u64) has copy, drop;
+    fun forty_two(): u64 {
+        42
+    }
+
+    public fun test(x: u64): u64 {
+        let f: | | | | u64 has drop = || || x;
+        f()();
+        let y = forty_two();
+        let f2: | | | | u64 has drop = || || y;
+        f2()();
```

Comment:
1. Could we add some assertions here, like `assert!(f2()() == 42);`?
2. Can we add a `//# run` command below, so that we exercise the execution of this function?

Reply:
done